pkg/controllers/awsloadbalancercontroller/ingressclass_test.go

```diff
@@ -0,0 +1,106 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"testing"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+func TestDesiredIngressClass(t *testing.T) {
+	ic := desiredIngressClass("test")
+	if ic.Name != "test" {
+		t.Errorf("unexpected name in desired ingress class, expected %q, got %q", "test", ic.Name)
+	}
+	if ic.Spec.Controller != albIngressClassController {
+		t.Errorf("unexpected controller in desired ingress class, expected %q, got %q", albIngressClassController, ic.Spec.Controller)
+	}
+}
+
+func TestEnsureIngressClass(t *testing.T) {
+	for _, tc := range []struct {
+		name                 string
+		existingIngressClass *networkingv1.IngressClass
+		ingressClassName     string
+		deletedIngressClass  bool
+	}{
+		{
+			name:             "no existing ingress class",
+			ingressClassName: "new",
+		},
+		{
+			name:                 "existing ingress class",
+			existingIngressClass: desiredIngressClass("old"),
+			ingressClassName:     "new",
+			deletedIngressClass:  true,
+		},
+		{
+			name:                 "existing ingress class, name no change",
+			existingIngressClass: desiredIngressClass("old"),
+			ingressClassName:     "old",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			var existingObjects []client.Object
+			if tc.existingIngressClass != nil {
+				existingObjects = append(existingObjects, tc.existingIngressClass)
+			}
+			controller := &albo.AWSLoadBalancerController{
+				ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
```

Comment:
I wanted to use a name other than the current "cluster" because I want to ensure that the logic is sound even when multiple instances of the controller will be supported. I've changed the name to _"test"_ so that this is clearly indicated.